File: clients/ruby/models/position.rb
class Position
  attr_accessor :account, :symbol, :underlying, :quote_currency, :currency, :commission, :prev_realised_pnl, :prev_unrealised_pnl, :prev_close_price, :realised_cost, :realised_pnl, :opening_timestamp, :opening_qty, :opening_cost, :opening_comm, :open_order_buy_qty, :open_order_buy_cost, :open_order_sell_qty, :open_order_sell_cost, :exec_buy_qty, :exec_buy_cost, :exec_sell_qty, :exec_sell_cost, :exec_qty, :exec_cost, :exec_comm, :current_timestamp, :current_qty, :current_cost, :current_comm, :unrealised_cost, :gross_open_cost, :gross_exec_cost, :last_price, :last_value, :home_notional, :foreign_notional, :init_margin, :maint_margin, :session_margin, :target_excess_margin, :var_margin, :unrealised_pnl, :unrealised_pnl_pcnt, :simple_qty, :simple_cost, :simple_value, :simple_pnl, :simple_pnl_pcnt, :avg_entry_price, :break_even_price, :margin_call_price, :liquidation_price, :bankrupt_price, :timestamp

  # :internal => :external
  def self.attribute_map
    {
      :account => :account,
      :symbol => :symbol,
      :underlying => :underlying,
      :quote_currency => :quoteCurrency,
      :currency => :currency,
      :commission => :commission,
      :prev_realised_pnl => :prevRealisedPnl,
      :prev_unrealised_pnl => :prevUnrealisedPnl,
      :prev_close_price => :prevClosePrice,
      :realised_cost => :realisedCost,
      :realised_pnl => :realisedPnl,
      :opening_timestamp => :openingTimestamp,
      :opening_qty => :openingQty,
      :opening_cost => :openingCost,
      :opening_comm => :openingComm,
      :open_order_buy_qty => :openOrderBuyQty,
      :open_order_buy_cost => :openOrderBuyCost,
      :open_order_sell_qty => :openOrderSellQty,
      :open_order_sell_cost => :openOrderSellCost,
      :exec_buy_qty => :execBuyQty,
      :exec_buy_cost => :execBuyCost,
      :exec_sell_qty => :execSellQty,
      :exec_sell_cost => :execSellCost,
      :exec_qty => :execQty,
      :exec_cost => :execCost,
      :exec_comm => :execComm,
      :current_timestamp => :currentTimestamp,
      :current_qty => :currentQty,
      :current_cost => :currentCost,
      :current_comm => :currentComm,
      :unrealised_cost => :unrealisedCost,
      :gross_open_cost => :grossOpenCost,
      :gross_exec_cost => :grossExecCost,
      :last_price => :lastPrice,
      :last_value => :lastValue,
      :home_notional => :homeNotional,
      :foreign_notional => :foreignNotional,
      :init_margin => :initMargin,
      :maint_margin => :maintMargin,
      :session_margin => :sessionMargin,
      :target_excess_margin => :targetExcessMargin,
      :var_margin => :varMargin,
      :unrealised_pnl => :unrealisedPnl,
      :unrealised_pnl_pcnt => :unrealisedPnlPcnt,
      :simple_qty => :simpleQty,
      :simple_cost => :simpleCost,
      :simple_value => :simpleValue,
      :simple_pnl => :simplePnl,
      :simple_pnl_pcnt => :simplePnlPcnt,
      :avg_entry_price => :avgEntryPrice,
      :break_even_price => :breakEvenPrice,
      :margin_call_price => :marginCallPrice,
      :liquidation_price => :liquidationPrice,
      :bankrupt_price => :bankruptPrice,
      :timestamp => :timestamp

    }
  end

  def initialize(attributes = {})
    return if attributes.empty?
    # Morph attribute keys into undescored rubyish style
    if self.class.attribute_map[:"account"]
      @account = attributes["account"]
    end
    if self.class.attribute_map[:"symbol"]
      @symbol = attributes["symbol"]
    end
    if self.class.attribute_map[:"underlying"]
      @underlying = attributes["underlying"]
    end
    if self.class.attribute_map[:"quote_currency"]
      @quote_currency = attributes["quoteCurrency"]
    end
    if self.class.attribute_map[:"currency"]
      @currency = attributes["currency"]
    end
    if self.class.attribute_map[:"commission"]
      @commission = attributes["commission"]
    end
    if self.class.attribute_map[:"prev_realised_pnl"]
      @prev_realised_pnl = attributes["prevRealisedPnl"]
    end
    if self.class.attribute_map[:"prev_unrealised_pnl"]
      @prev_unrealised_pnl = attributes["prevUnrealisedPnl"]
    end
    if self.class.attribute_map[:"prev_close_price"]
      @prev_close_price = attributes["prevClosePrice"]
    end
    if self.class.attribute_map[:"realised_cost"]
      @realised_cost = attributes["realisedCost"]
    end
    if self.class.attribute_map[:"realised_pnl"]
      @realised_pnl = attributes["realisedPnl"]
    end
    if self.class.attribute_map[:"opening_timestamp"]
      @opening_timestamp = attributes["openingTimestamp"]
    end
    if self.class.attribute_map[:"opening_qty"]
      @opening_qty = attributes["openingQty"]
    end
    if self.class.attribute_map[:"opening_cost"]
      @opening_cost = attributes["openingCost"]
    end
    if self.class.attribute_map[:"opening_comm"]
      @opening_comm = attributes["openingComm"]
    end
    if self.class.attribute_map[:"open_order_buy_qty"]
      @open_order_buy_qty = attributes["openOrderBuyQty"]
    end
    if self.class.attribute_map[:"open_order_buy_cost"]
      @open_order_buy_cost = attributes["openOrderBuyCost"]
    end
    if self.class.attribute_map[:"open_order_sell_qty"]
      @open_order_sell_qty = attributes["openOrderSellQty"]
    end
    if self.class.attribute_map[:"open_order_sell_cost"]
      @open_order_sell_cost = attributes["openOrderSellCost"]
    end
    if self.class.attribute_map[:"exec_buy_qty"]
      @exec_buy_qty = attributes["execBuyQty"]
    end
    if self.class.attribute_map[:"exec_buy_cost"]
      @exec_buy_cost = attributes["execBuyCost"]
    end
    if self.class.attribute_map[:"exec_sell_qty"]
      @exec_sell_qty = attributes["execSellQty"]
    end
    if self.class.attribute_map[:"exec_sell_cost"]
      @exec_sell_cost = attributes["execSellCost"]
    end
    if self.class.attribute_map[:"exec_qty"]
      @exec_qty = attributes["execQty"]
    end
    if self.class.attribute_map[:"exec_cost"]
      @exec_cost = attributes["execCost"]
    end
    if self.class.attribute_map[:"exec_comm"]
      @exec_comm = attributes["execComm"]
    end
    if self.class.attribute_map[:"current_timestamp"]
      @current_timestamp = attributes["currentTimestamp"]
    end
    if self.class.attribute_map[:"current_qty"]
      @current_qty = attributes["currentQty"]
    end
    if self.class.attribute_map[:"current_cost"]
      @current_cost = attributes["currentCost"]
    end
    if self.class.attribute_map[:"current_comm"]
      @current_comm = attributes["currentComm"]
    end
    if self.class.attribute_map[:"unrealised_cost"]
      @unrealised_cost = attributes["unrealisedCost"]
    end
    if self.class.attribute_map[:"gross_open_cost"]
      @gross_open_cost = attributes["grossOpenCost"]
    end
    if self.class.attribute_map[:"gross_exec_cost"]
      @gross_exec_cost = attributes["grossExecCost"]
    end
    if self.class.attribute_map[:"last_price"]
      @last_price = attributes["lastPrice"]
    end
    if self.class.attribute_map[:"last_value"]
      @last_value = attributes["lastValue"]
    end
    if self.class.attribute_map[:"home_notional"]
      @home_notional = attributes["homeNotional"]
    end
    if self.class.attribute_map[:"foreign_notional"]
      @foreign_notional = attributes["foreignNotional"]
    end
    if self.class.attribute_map[:"init_margin"]
      @init_margin = attributes["initMargin"]
    end
    if self.class.attribute_map[:"maint_margin"]
      @maint_margin = attributes["maintMargin"]
    end
    if self.class.attribute_map[:"session_margin"]
      @session_margin = attributes["sessionMargin"]
    end
    if self.class.attribute_map[:"target_excess_margin"]
      @target_excess_margin = attributes["targetExcessMargin"]
    end
    if self.class.attribute_map[:"var_margin"]
      @var_margin = attributes["varMargin"]
    end
    if self.class.attribute_map[:"unrealised_pnl"]
      @unrealised_pnl = attributes["unrealisedPnl"]
    end
    if self.class.attribute_map[:"unrealised_pnl_pcnt"]
      @unrealised_pnl_pcnt = attributes["unrealisedPnlPcnt"]
    end
    if self.class.attribute_map[:"simple_qty"]
      @simple_qty = attributes["simpleQty"]
    end
    if self.class.attribute_map[:"simple_cost"]
      @simple_cost = attributes["simpleCost"]
    end
    if self.class.attribute_map[:"simple_value"]
      @simple_value = attributes["simpleValue"]
    end
    if self.class.attribute_map[:"simple_pnl"]
      @simple_pnl = attributes["simplePnl"]
    end
    if self.class.attribute_map[:"simple_pnl_pcnt"]
      @simple_pnl_pcnt = attributes["simplePnlPcnt"]
    end
    if self.class.attribute_map[:"avg_entry_price"]
      @avg_entry_price = attributes["avgEntryPrice"]
    end
    if self.class.attribute_map[:"break_even_price"]
      @break_even_price = attributes["breakEvenPrice"]
    end
    if self.class.attribute_map[:"margin_call_price"]
      @margin_call_price = attributes["marginCallPrice"]
    end
    if self.class.attribute_map[:"liquidation_price"]
      @liquidation_price = attributes["liquidationPrice"]
    end
    if self.class.attribute_map[:"bankrupt_price"]
      @bankrupt_price = attributes["bankruptPrice"]
    end
    if self.class.attribute_map[:"timestamp"]
      @timestamp = attributes["timestamp"]
    end
    

  end

  def to_body
    body = {}
    self.class.attribute_map.each_pair do |key, value|
      body[value] = self.send(key) unless self.send(key).nil?
    end
    body
  end
end

File: clients/java/android/src/main/java/com/wordnik/client/model/Position.java
package com.wordnik.client.model;

import com.fasterxml.jackson.annotation.JsonProperty;

import java.util.Date;
public class Position {
  @JsonProperty("account")
  private Double account = null;
  @JsonProperty("symbol")
  private String symbol = null;
  @JsonProperty("currency")
  private String currency = null;
  @JsonProperty("commission")
  private Double commission = null;
  @JsonProperty("prevRealisedPnl")
  private Double prevRealisedPnl = null;
  @JsonProperty("prevUnrealisedPnl")
  private Double prevUnrealisedPnl = null;
  @JsonProperty("prevClosePrice")
  private Double prevClosePrice = null;
  @JsonProperty("realisedCost")
  private Double realisedCost = null;
  @JsonProperty("realisedPnl")
  private Double realisedPnl = null;
  @JsonProperty("openingTimestamp")
  private Date openingTimestamp = null;
  @JsonProperty("openingQty")
  private Double openingQty = null;
  @JsonProperty("openingCost")
  private Double openingCost = null;
  @JsonProperty("openingComm")
  private Double openingComm = null;
  @JsonProperty("openOrderBuyQty")
  private Double openOrderBuyQty = null;
  @JsonProperty("openOrderBuyCost")
  private Double openOrderBuyCost = null;
  @JsonProperty("openOrderSellQty")
  private Double openOrderSellQty = null;
  @JsonProperty("openOrderSellCost")
  private Double openOrderSellCost = null;
  @JsonProperty("execBuyQty")
  private Double execBuyQty = null;
  @JsonProperty("execBuyCost")
  private Double execBuyCost = null;
  @JsonProperty("execSellQty")
  private Double execSellQty = null;
  @JsonProperty("execSellCost")
  private Double execSellCost = null;
  @JsonProperty("execQty")
  private Double execQty = null;
  @JsonProperty("execCost")
  private Double execCost = null;
  @JsonProperty("execComm")
  private Double execComm = null;
  @JsonProperty("currentTimestamp")
  private Date currentTimestamp = null;
  @JsonProperty("currentQty")
  private Double currentQty = null;
  @JsonProperty("currentCost")
  private Double currentCost = null;
  @JsonProperty("currentComm")
  private Double currentComm = null;
  @JsonProperty("unrealisedCost")
  private Double unrealisedCost = null;
  @JsonProperty("grossOpenCost")
  private Double grossOpenCost = null;
  @JsonProperty("grossExecCost")
  private Double grossExecCost = null;
  @JsonProperty("lastPrice")
  private Double lastPrice = null;
  @JsonProperty("lastValue")
  private Double lastValue = null;
  @JsonProperty("notionalCurrency1")
  private String notionalCurrency1 = null;
  @JsonProperty("notionalValue1")
  private Double notionalValue1 = null;
  @JsonProperty("notionalCurrency2")
  private String notionalCurrency2 = null;
  @JsonProperty("notionalValue2")
  private Double notionalValue2 = null;
  @JsonProperty("initMargin")
  private Double initMargin = null;
  @JsonProperty("maintMargin")
  private Double maintMargin = null;
  @JsonProperty("sessionMargin")
  private Double sessionMargin = null;
  @JsonProperty("targetExcessMargin")
  private Double targetExcessMargin = null;
  @JsonProperty("varMargin")
  private Double varMargin = null;
  @JsonProperty("unrealisedPnl")
  private Double unrealisedPnl = null;
  @JsonProperty("unrealisedPnlPcnt")
  private Double unrealisedPnlPcnt = null;
  @JsonProperty("avgEntryPrice")
  private Double avgEntryPrice = null;
  @JsonProperty("breakEvenPrice")
  private Double breakEvenPrice = null;
  @JsonProperty("marginCallPrice")
  private Double marginCallPrice = null;
  @JsonProperty("liquidationPrice")
  private Double liquidationPrice = null;
  @JsonProperty("bankruptPrice")
  private Double bankruptPrice = null;
  @JsonProperty("timestamp")
  private Date timestamp = null;
  public Double getAccount() {
    return account;
  }
  public void setAccount(Double account) {
    this.account = account;
  }

  public String getSymbol() {
    return symbol;
  }
  public void setSymbol(String symbol) {
    this.symbol = symbol;
  }

  public String getCurrency() {
    return currency;
  }
  public void setCurrency(String currency) {
    this.currency = currency;
  }

  public Double getCommission() {
    return commission;
  }
  public void setCommission(Double commission) {
    this.commission = commission;
  }

  public Double getPrevRealisedPnl() {
    return prevRealisedPnl;
  }
  public void setPrevRealisedPnl(Double prevRealisedPnl) {
    this.prevRealisedPnl = prevRealisedPnl;
  }

  public Double getPrevUnrealisedPnl() {
    return prevUnrealisedPnl;
  }
  public void setPrevUnrealisedPnl(Double prevUnrealisedPnl) {
    this.prevUnrealisedPnl = prevUnrealisedPnl;
  }

  public Double getPrevClosePrice() {
    return prevClosePrice;
  }
  public void setPrevClosePrice(Double prevClosePrice) {
    this.prevClosePrice = prevClosePrice;
  }

  public Double getRealisedCost() {
    return realisedCost;
  }
  public void setRealisedCost(Double realisedCost) {
    this.realisedCost = realisedCost;
  }

  public Double getRealisedPnl() {
    return realisedPnl;
  }
  public void setRealisedPnl(Double realisedPnl) {
    this.realisedPnl = realisedPnl;
  }

  public Date getOpeningTimestamp() {
    return openingTimestamp;
  }
  public void setOpeningTimestamp(Date openingTimestamp) {
    this.openingTimestamp = openingTimestamp;
  }

  public Double getOpeningQty() {
    return openingQty;
  }
  public void setOpeningQty(Double openingQty) {
    this.openingQty = openingQty;
  }

  public Double getOpeningCost() {
    return openingCost;
  }
  public void setOpeningCost(Double openingCost) {
    this.openingCost = openingCost;
  }

  public Double getOpeningComm() {
    return openingComm;
  }
  public void setOpeningComm(Double openingComm) {
    this.openingComm = openingComm;
  }

  public Double getOpenOrderBuyQty() {
    return openOrderBuyQty;
  }
  public void setOpenOrderBuyQty(Double openOrderBuyQty) {
    this.openOrderBuyQty = openOrderBuyQty;
  }

  public Double getOpenOrderBuyCost() {
    return openOrderBuyCost;
  }
  public void setOpenOrderBuyCost(Double openOrderBuyCost) {
    this.openOrderBuyCost = openOrderBuyCost;
  }

  public Double getOpenOrderSellQty() {
    return openOrderSellQty;
  }
  public void setOpenOrderSellQty(Double openOrderSellQty) {
    this.openOrderSellQty = openOrderSellQty;
  }

  public Double getOpenOrderSellCost() {
    return openOrderSellCost;
  }
  public void setOpenOrderSellCost(Double openOrderSellCost) {
    this.openOrderSellCost = openOrderSellCost;
  }

  public Double getExecBuyQty() {
    return execBuyQty;
  }
  public void setExecBuyQty(Double execBuyQty) {
    this.execBuyQty = execBuyQty;
  }

  public Double getExecBuyCost() {
    return execBuyCost;
  }
  public void setExecBuyCost(Double execBuyCost) {
    this.execBuyCost = execBuyCost;
  }

  public Double getExecSellQty() {
    return execSellQty;
  }
  public void setExecSellQty(Double execSellQty) {
    this.execSellQty = execSellQty;
  }

  public Double getExecSellCost() {
    return execSellCost;
  }
  public void setExecSellCost(Double execSellCost) {
    this.execSellCost = execSellCost;
  }

  public Double getExecQty() {
    return execQty;
  }
  public void setExecQty(Double execQty) {
    this.execQty = execQty;
  }

  public Double getExecCost() {
    return execCost;
  }
  public void setExecCost(Double execCost) {
    this.execCost = execCost;
  }

  public Double getExecComm() {
    return execComm;
  }
  public void setExecComm(Double execComm) {
    this.execComm = execComm;
  }

  public Date getCurrentTimestamp() {
    return currentTimestamp;
  }
  public void setCurrentTimestamp(Date currentTimestamp) {
    this.currentTimestamp = currentTimestamp;
  }

  public Double getCurrentQty() {
    return currentQty;
  }
  public void setCurrentQty(Double currentQty) {
    this.currentQty = currentQty;
  }

  public Double getCurrentCost() {
    return currentCost;
  }
  public void setCurrentCost(Double currentCost) {
    this.currentCost = currentCost;
  }

  public Double getCurrentComm() {
    return currentComm;
  }
  public void setCurrentComm(Double currentComm) {
    this.currentComm = currentComm;
  }

  public Double getUnrealisedCost() {
    return unrealisedCost;
  }
  public void setUnrealisedCost(Double unrealisedCost) {
    this.unrealisedCost = unrealisedCost;
  }

  public Double getGrossOpenCost() {
    return grossOpenCost;
  }
  public void setGrossOpenCost(Double grossOpenCost) {
    this.grossOpenCost = grossOpenCost;
  }

  public Double getGrossExecCost() {
    return grossExecCost;
  }
  public void setGrossExecCost(Double grossExecCost) {
    this.grossExecCost = grossExecCost;
  }

  public Double getLastPrice() {
    return lastPrice;
  }
  public void setLastPrice(Double lastPrice) {
    this.lastPrice = lastPrice;
  }

  public Double getLastValue() {
    return lastValue;
  }
  public void setLastValue(Double lastValue) {
    this.lastValue = lastValue;
  }

  public String getNotionalCurrency1() {
    return notionalCurrency1;
  }
  public void setNotionalCurrency1(String notionalCurrency1) {
    this.notionalCurrency1 = notionalCurrency1;
  }

  public Double getNotionalValue1() {
    return notionalValue1;
  }
  public void setNotionalValue1(Double notionalValue1) {
    this.notionalValue1 = notionalValue1;
  }

  public String getNotionalCurrency2() {
    return notionalCurrency2;
  }
  public void setNotionalCurrency2(String notionalCurrency2) {
    this.notionalCurrency2 = notionalCurrency2;
  }

  public Double getNotionalValue2() {
    return notionalValue2;
  }
  public void setNotionalValue2(Double notionalValue2) {
    this.notionalValue2 = notionalValue2;
  }

  public Double getInitMargin() {
    return initMargin;
  }
  public void setInitMargin(Double initMargin) {
    this.initMargin = initMargin;
  }

  public Double getMaintMargin() {
    return maintMargin;
  }
  public void setMaintMargin(Double maintMargin) {
    this.maintMargin = maintMargin;
  }

  public Double getSessionMargin() {
    return sessionMargin;
  }
  public void setSessionMargin(Double sessionMargin) {
    this.sessionMargin = sessionMargin;
  }

  public Double getTargetExcessMargin() {
    return targetExcessMargin;
  }
  public void setTargetExcessMargin(Double targetExcessMargin) {
    this.targetExcessMargin = targetExcessMargin;
  }

  public Double getVarMargin() {
    return varMargin;
  }
  public void setVarMargin(Double varMargin) {
    this.varMargin = varMargin;
  }

  public Double getUnrealisedPnl() {
    return unrealisedPnl;
  }
  public void setUnrealisedPnl(Double unrealisedPnl) {
    this.unrealisedPnl = unrealisedPnl;
  }

  public Double getUnrealisedPnlPcnt() {
    return unrealisedPnlPcnt;
  }
  public void setUnrealisedPnlPcnt(Double unrealisedPnlPcnt) {
    this.unrealisedPnlPcnt = unrealisedPnlPcnt;
  }

  public Double getAvgEntryPrice() {
    return avgEntryPrice;
  }
  public void setAvgEntryPrice(Double avgEntryPrice) {
    this.avgEntryPrice = avgEntryPrice;
  }

  public Double getBreakEvenPrice() {
    return breakEvenPrice;
  }
  public void setBreakEvenPrice(Double breakEvenPrice) {
    this.breakEvenPrice = breakEvenPrice;
  }

  public Double getMarginCallPrice() {
    return marginCallPrice;
  }
  public void setMarginCallPrice(Double marginCallPrice) {
    this.marginCallPrice = marginCallPrice;
  }

  public Double getLiquidationPrice() {
    return liquidationPrice;
  }
  public void setLiquidationPrice(Double liquidationPrice) {
    this.liquidationPrice = liquidationPrice;
  }

  public Double getBankruptPrice() {
    return bankruptPrice;
  }
  public void setBankruptPrice(Double bankruptPrice) {
    this.bankruptPrice = bankruptPrice;
  }

  public Date getTimestamp() {
    return timestamp;
  }
  public void setTimestamp(Date timestamp) {
    this.timestamp = timestamp;
  }

  @Override
  public String toString()  {
    StringBuilder sb = new StringBuilder();
    sb.append("class Position {\n");
    sb.append("  account: ").append(account).append("\n");
    sb.append("  symbol: ").append(symbol).append("\n");
    sb.append("  currency: ").append(currency).append("\n");
    sb.append("  commission: ").append(commission).append("\n");
    sb.append("  prevRealisedPnl: ").append(prevRealisedPnl).append("\n");
    sb.append("  prevUnrealisedPnl: ").append(prevUnrealisedPnl).append("\n");
    sb.append("  prevClosePrice: ").append(prevClosePrice).append("\n");
    sb.append("  realisedCost: ").append(realisedCost).append("\n");
    sb.append("  realisedPnl: ").append(realisedPnl).append("\n");
    sb.append("  openingTimestamp: ").append(openingTimestamp).append("\n");
    sb.append("  openingQty: ").append(openingQty).append("\n");
    sb.append("  openingCost: ").append(openingCost).append("\n");
    sb.append("  openingComm: ").append(openingComm).append("\n");
    sb.append("  openOrderBuyQty: ").append(openOrderBuyQty).append("\n");
    sb.append("  openOrderBuyCost: ").append(openOrderBuyCost).append("\n");
    sb.append("  openOrderSellQty: ").append(openOrderSellQty).append("\n");
    sb.append("  openOrderSellCost: ").append(openOrderSellCost).append("\n");
    sb.append("  execBuyQty: ").append(execBuyQty).append("\n");
    sb.append("  execBuyCost: ").append(execBuyCost).append("\n");
    sb.append("  execSellQty: ").append(execSellQty).append("\n");
    sb.append("  execSellCost: ").append(execSellCost).append("\n");
    sb.append("  execQty: ").append(execQty).append("\n");
    sb.append("  execCost: ").append(execCost).append("\n");
    sb.append("  execComm: ").append(execComm).append("\n");
    sb.append("  currentTimestamp: ").append(currentTimestamp).append("\n");
    sb.append("  currentQty: ").append(currentQty).append("\n");
    sb.append("  currentCost: ").append(currentCost).append("\n");
    sb.append("  currentComm: ").append(currentComm).append("\n");
    sb.append("  unrealisedCost: ").append(unrealisedCost).append("\n");
    sb.append("  grossOpenCost: ").append(grossOpenCost).append("\n");
    sb.append("  grossExecCost: ").append(grossExecCost).append("\n");
    sb.append("  lastPrice: ").append(lastPrice).append("\n");
    sb.append("  lastValue: ").append(lastValue).append("\n");
    sb.append("  notionalCurrency1: ").append(notionalCurrency1).append("\n");
    sb.append("  notionalValue1: ").append(notionalValue1).append("\n");
    sb.append("  notionalCurrency2: ").append(notionalCurrency2).append("\n");
    sb.append("  notionalValue2: ").append(notionalValue2).append("\n");
    sb.append("  initMargin: ").append(initMargin).append("\n");
    sb.append("  maintMargin: ").append(maintMargin).append("\n");
    sb.append("  sessionMargin: ").append(sessionMargin).append("\n");
    sb.append("  targetExcessMargin: ").append(targetExcessMargin).append("\n");
    sb.append("  varMargin: ").append(varMargin).append("\n");
    sb.append("  unrealisedPnl: ").append(unrealisedPnl).append("\n");
    sb.append("  unrealisedPnlPcnt: ").append(unrealisedPnlPcnt).append("\n");
    sb.append("  avgEntryPrice: ").append(avgEntryPrice).append("\n");
    sb.append("  breakEvenPrice: ").append(breakEvenPrice).append("\n");
    sb.append("  marginCallPrice: ").append(marginCallPrice).append("\n");
    sb.append("  liquidationPrice: ").append(liquidationPrice).append("\n");
    sb.append("  bankruptPrice: ").append(bankruptPrice).append("\n");
    sb.append("  timestamp: ").append(timestamp).append("\n");
    sb.append("}\n");
    return sb.toString();
  }
}

